### ISSUE TITLE
Produce Schema design doc for packages/projects

### ISSUE THREAD

**Issue by eeyun** [habitat-sh/builder#1183]
Issue #1177 tracks work for new plan connection and visibility flow.

As a part of that work we're going to need to make adjustments to the way in which we handle packages and projects in the backend. 

In order to ensure that we aren't creating more technical debt for ourselves we should produce a design doc for the proposed schema changes before any work is implemented.

**Comment by eeyun** [comment]
Notes for later: visibility should get removed from the origin_projects and target needs to get added

**Comment by eeyun** [comment]
Potentially add a New api for connectable targets?
